### PR TITLE
builtins: implement regexp_split_to{table,array}

### PR DESCRIPTION
**Pull request by otan**
Refs: #51818

Release note (sql change): Implement the regexp_split_to_table and
regexp_split_to_array builtins.

### CODE REVIEW

File: pkg/sql/logictest/testdata/logic_test/builtin_function [hunk 1]
```diff
@@ -2786,9 +2790,32 @@ SELECT (SELECT * FROM [SHOW crdb_version]) = version()
...
-
 query B
 select crdb_internal.json_to_pb('cockroach.sql.sqlbase.Descriptor', crdb_internal.pb_to_json('cockroach.sql.sqlbase.Descriptor', descriptor)) = descriptor from system.descriptor where id = 1
 ----
 true
 
+subtest regexp_split
+
+query T
+SELECT regexp_split_to_table('3a0A1', 'a')
```
Comment (rohany) [inline pkg/sql/logictest/testdata/logic_test/builtin_function:2801]: can you test with some regexes

File: pkg/sql/sem/builtins/builtins.go [hunk 2]
```diff
@@ -1437,26 +1458,56 @@ var builtins = map[string]builtinDefinition{
...
+	"regexp_split_to_array": makeBuiltin(
+		defProps(),
+		tree.Overload{
+			Types: tree.ArgTypes{
+				{"string", types.String},
+				{"pattern", types.String},
+			},
+			ReturnType: tree.FixedReturnType(types.MakeArray(types.String)),
+			Fn: func(ctx *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				words, err := regexpSplit(ctx, args, false /* hasFlags */)
```
Comment (rohany) [inline pkg/sql/sem/builtins/builtins.go:1475]: can probably make this function body a short helper

File: pkg/sql/logictest/testdata/logic_test/builtin_function (hunk 1 above)
Comment (otan) [reply pkg/sql/logictest/testdata/logic_test/builtin_function]: Done!

File: pkg/sql/sem/builtins/builtins.go (hunk 2 above)
Comment (otan) [reply pkg/sql/sem/builtins/builtins.go]: Done!

**Review by rohany — APPROVED** [review]

**Comment by otan** [comment]
bors r=rohany

**Pull request merged**